File: eukay/lib/pages/dashboard/mappers/review_model.dart
class ReviewModel {
  final int starRating;
  final String review, productName, userName, userImage, productImage, id;

  const ReviewModel({
    required this.id,
    required this.starRating,
    required this.review,
    required this.userName,
    required this.userImage,
    required this.productName,
    required this.productImage,
  });

  factory ReviewModel.fromJson(Map<String, dynamic> json) {
    return ReviewModel(
      id: json["_id"],
      starRating: json["starRating"],
      review: json["review"],
      userName: json["userName"],
      userImage: json["userImage"],
      productName: json["productName"],
      productImage: json["productImage"],
    );
  }
}

File: server/models/review.model.js
const mongoose = require("mongoose");

const ReviewSchema = mongoose.Schema(
  {
    starRating: {
      type: Number,
      required: true,
    },

    review: {
      type: String,
      required: true,
    },

    productId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Product",
      required: true,
    },

    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },

    productImage: {
      type: String,
      required: true,
    },

    productName: {
      type: String,
      required: true,
    },

    sellerId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Seller",
      required: true,
    },
  },
  {
    timestamps: true,
  }
);

const Review = mongoose.model("Review", ReviewSchema);

module.exports = Review;
